raiden_contracts/deploy/contract_deployer.py

```diff
@@ -211,6 +212,65 @@ def _deploy_and_remember(
             address=deployed_contracts['contracts'][contract_name]['address'],
         )
 
+    def register_token_network(
+            self,
+            token_registry_abi: Dict,
+            token_registry_address: str,
+            token_address: str,
+            channel_participant_deposit_limit: Optional[int],
+            token_network_deposit_limit: Optional[int],
+    ):
+        """Register token with a TokenNetworkRegistry contract."""
+        with_limits = contracts_version_expects_deposit_limits(self.contracts_version)
+        if with_limits:
+            if channel_participant_deposit_limit is None:
+                raise ValueError(
+                    'contracts_version 0.9.0 and afterwards expect '
+                    'channel_participant_deposit_limit',
+                )
+            if token_network_deposit_limit is None:
+                raise ValueError(
+                    'contracts_version 0.9.0 and afterwards expect '
+                    'token_network_deposit_limit',
+                )
+        else:
+            if channel_participant_deposit_limit:
+                raise ValueError(
+                    'contracts_version below 0.9.0 does not expect '
+                    'channel_participant_deposit_limit',
+                )
+            if token_network_deposit_limit:
+                raise ValueError(
+                    'contracts_version below 0.9.0 does not expect token_network_deposit_limit',
+                )
+        token_network_registry = self.web3.eth.contract(
+            abi=token_registry_abi,
+            address=token_registry_address,
+        )
+
+        version_from_onchain = token_network_registry.functions.contract_version().call()
+        assert version_from_onchain == self.contract_manager.version_string(), \
+            f'got {version_from_onchain}, expected {self.contract_manager.version_string()}'
+
+        if with_limits:
+            command = token_network_registry.functions.createERC20TokenNetwork(
+                _token_address=token_address,
+                _channel_participant_deposit_limit=channel_participant_deposit_limit,
+                _token_network_deposit_limit=token_network_deposit_limit,
```

Comment:
Out of curiosity, what's with the leading underscore in the parameter name?

Reply:
Some Solidity coders use `_arg` for arguments.  For example https://www.ethereum.org/greeter#your-first-citizen-the-greeter .  I'm just keeping the convention.